### DISCUSSION TITLE
Product Default Quantity

### DISCUSSION THREAD

**Discussion by tristancliffe** [Shopify/dawn#3199] [Q&A]
I have been updating my lightly customised Dawn theme from 7.0 to 12.0 to make use of new features, bug fixes etc, but I am stumped on the Quantity selector in products. I don't have the B2B options for catalogs as I am not on that Shopify plan.

In V7.0 I had a section of code that used a product tag to set the 'default' quantity value, but in 12.0 I cannot make that work, despite replacing the whole section of code in the quantity selector section. This is in main_product.liquid.
```

{%- when 'quantity_selector' -%}
              <div
                class="product-form__input product-form__quantity{% if settings.inputs_shadow_vertical_offset != 0 and settings.inputs_shadow_vertical_offset < 0 %} product-form__quantity-top{% endif %}"
                {{ block.shopify_attributes }}
              >
                <label class="form__label" for="Quantity-{{ section.id }}">
                  {{ 'products.product.quantity.label' | t }}
                </label>
                {% assign DefaultQty = "1" %}
                {% for tag in product.tags %}
                    {% if tag contains 'DefaultQuantity:' %}
                        {% assign DefaultQty = tag | remove: 'DefaultQuantity:' %}
                        {% break %}
                    {% endif %}
                {% endfor %}
                <quantity-input class="quantity">
                  <button class="quantity__button no-js-hidden" name="minus" type="button">
                    <span class="visually-hidden">
                      {{- 'products.product.quantity.decrease' | t: product: product.title | escape -}}
                    </span>
                    {% render 'icon-minus' %}
                  </button>
                  <input
                    class="quantity__input"
                    type="number"
                    name="quantity"
                    id="Quantity-{{ section.id }}"
                    min="1"
                    value="{{ DefaultQty }}"
                    form="{{ product_form_id }}"
                  >
                  <button class="quantity__button no-js-hidden" name="plus" type="button">
                    <span class="visually-hidden">
                      {{- 'products.product.quantity.increase' | t: product: product.title | escape -}}
                    </span>
                    {% render 'icon-plus' %}
                  </button>
                </quantity-input>
              </div>
   
```           
What would be the best v12.0 way of using the DefaultQuantity tag to drive the initial value in the quantity selector these days, again bearing in mind I don't use the B2B functions (and I'm aware that Dawn is meant to work with stuff like that - this isn't a complaint about fancy features!)?

Thanks.

**Comment by rsusano** [comment]
Hi! In Dawn 12, the quantity selector ultimately renders an actual numeric `<input>` (so the simplest approach is to set its initial `value` from your tag).

In `sections/main-product.liquid` (quantity selector block), replace the hardcoded default with:

{% assign DefaultQty = 1 %}
{% for tag in product.tags %}
  {% if tag contains 'DefaultQuantity:' %}
    {% assign DefaultQty = tag | remove: 'DefaultQuantity:' | plus: 0 %}
    {% break %}
  {% endif %}
{% endfor %}

Then set the quantity input to:

value="{{ DefaultQty }}"

That will drive the initial quantity from your `product.tags` without relying on B2B/catalog features.
